### ISSUE TITLE
Inconsistency in Model Checkpoints

### ISSUE THREAD

**Issue by WeiHan00457030** [MIC-DKFZ/nnUNet#2350]
During the training, I save the model checkpoint every 50 epochs. In the training log, it shows that the training loss is continuously changing and the best model is consistently being updated. However, when I compare the network weights of two checkpoints, I find that they are exactly the same. When I specify a checkpoint for inference, I also get exactly the same results.
